core/iwasm/interpreter/wasm_loader.c

```diff
@@ -1163,7 +1163,11 @@ load_memory_import(const uint8 **p_buf, const uint8 *buf_end,
             (WASMModuleCommon *)parent_module, sub_module_name, error_buf,
             error_buf_size);
         if (!sub_module) {
+#if WASM_ENABLE_LIB_WASI_THREADS != 0
+            goto set_memory;
```

Comment:
Using goto here may be not a good coding style, and could we skip the check only when wasi-threads is enabled and the memory is shared?
```C
        if (!sub_module) {
#if WASM_ENABLE_LIB_WASI_THREADS != 0
            /* Avoid memory import failure when wasi-threads is enabled
               and the memory is shared */
            if (!(declare_max_page_count_flag & 2)
                return false;
#else
            return false;
#endif
        }
        else {
            linked_memory = wasm_loader_resolve_memory(
                sub_module_name, memory_name, declare_init_page_count,
                declare_max_page_count, error_buf, error_buf_size);
            if (!linked_memory) {
                return false;
            }
            ...
        }
```

Reply:
Thanks, that's better, I amended the commit with the suggested change.
It doesn't look like there's any change needed for AOT.